plugins/inputs/opensearch_query/opensearch_query.go

```diff
@@ -0,0 +1,326 @@
+type osAggregation struct {
+	Index             string          `toml:"index"`
+	MeasurementName   string          `toml:"measurement_name"`
+	DateField         string          `toml:"date_field"`
+	DateFieldFormat   string          `toml:"date_field_custom_format"`
+	QueryPeriod       config.Duration `toml:"query_period"`
+	FilterQuery       string          `toml:"filter_query"`
+	MetricFields      []string        `toml:"metric_fields"`
+	MetricFunction    string          `toml:"metric_function"`
+	Tags              []string        `toml:"tags"`
+	IncludeMissingTag bool            `toml:"include_missing_tag"`
+	MissingTagValue   string          `toml:"missing_tag_value"`
+	mapMetricFields   map[string]string
+
+	aggregation AggregationRequest
+}
+
+type mapping map[string]fieldIndex
+
+type fieldIndex struct {
+	Mappings map[string]fieldMapping `json:"mappings"`
+}
+
+type fieldMapping struct {
+	FullName string               `json:"full_name"`
+	Mapping  map[string]fieldType `json:"mapping"`
+}
+
+type fieldType struct {
+	Type string `json:"type"`
+}
+
+func (*OpensearchQuery) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init the plugin.
+func (o *OpensearchQuery) Init() error {
+	if o.URLs == nil {
+		return fmt.Errorf("no urls defined")
+	}
+
+	err := o.connectToOpensearch()
+	if err != nil {
+		o.Log.Warnf("error connecting to opensearch: %w", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(o.Timeout))
+	defer cancel()
+
+	for i, agg := range o.Aggregations {
+		if agg.MeasurementName == "" {
+			return fmt.Errorf("field 'measurement_name' is not set")
+		}
+		if agg.DateField == "" {
+			return fmt.Errorf("field 'date_field' is not set")
+		}
+		err = o.initAggregation(ctx, agg, i)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *OpensearchQuery) initAggregation(ctx context.Context, agg osAggregation, i int) (err error) {
+	// retrieve field mapping and build queries only once
+	agg.mapMetricFields, err = o.getMetricFields(ctx, agg)
+	if err != nil {
+		return fmt.Errorf("not possible to retrieve fields: %v", err.Error())
+	}
+
+	for _, metricField := range agg.MetricFields {
+		if _, ok := agg.mapMetricFields[metricField]; !ok {
+			return fmt.Errorf("metric field '%s' not found on index '%s'", metricField, agg.Index)
+		}
+	}
+
+	err = agg.buildAggregationQuery()
+	if err != nil {
+		return fmt.Errorf("error building aggregation: %s", err)
+	}
+
+	o.Aggregations[i] = agg
+	return nil
+}
+
+func (o *OpensearchQuery) connectToOpensearch() error {
+	var client *opensearch.Client
+	var transport *http.Transport
+
+	if o.InsecureSkipVerify {
+		transport = &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+	}
+
+	clientConfig := opensearch.Config{
+		Addresses: o.URLs,
+		Username:  o.Username,
+		Password:  o.Password,
+	}
+
+	if transport != nil {
+		clientConfig.Transport = transport
+	}
+
+	client, err := opensearch.NewClient(clientConfig)
```

Comment:
Why not just
```suggestion
	clientConfig := opensearch.Config{
		Addresses: o.URLs,
		Username:  o.Username,
		Password:  o.Password,
	}

	if o.InsecureSkipVerify {
		clientConfig.Transport  = &http.Transport{
			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
		}
	}

	client, err := opensearch.NewClient(clientConfig)
```

as there is no other way the transport cannot be `nil`...